prime-router/src/main/kotlin/azure/Event.kt

```diff
@@ -75,23 +82,90 @@ abstract class Event(val eventAction: EventAction, val at: OffsetDateTime?) {
     companion object {
         fun parseQueueMessage(event: String): Event {
             val parts = event.split(messageDelimiter)
-            if (parts.size < 3 || parts.size > 4) error("Internal Error: bad event format")
+            if (parts.size < 3 || parts.size > 7) error("Internal Error: bad event format")
             val action = EventAction.parseQueueMessage(parts[1])
-            val after = parts.getOrNull(3)?.let { OffsetDateTime.parse(it) }
             return when (parts[0]) {
                 ReportEvent.eventType -> {
+                    val after = parts.getOrNull(3)?.let { OffsetDateTime.parse(it) }
                     val reportId = UUID.fromString(parts[2])
                     ReportEvent(action, reportId, after)
                 }
                 ReceiverEvent.eventType -> {
+                    val after = parts.getOrNull(3)?.let { OffsetDateTime.parse(it) }
                     ReceiverEvent(action, parts[2], after)
                 }
+                ProcessEvent.eventType -> {
+                    val after = parts.getOrNull(6)?.let { OffsetDateTime.parse(it) }
```

Comment:
Should you check for exceptions on the parse here?  This code also is a duplicate as above.  maybe factor this out of the when.

Reply:
For processEvent there are 6 parts to the message, not 3.
There should be no way for exceptions since the message is serialized by ProcessEvent.toQueueMessage

Reply:
I think there's a problem here. As an example the case where a `default` is not passed but an `at` is results in what should be the `after` being in slot 5.

Reply:
Updated.